src/utils/context.rs

```diff
@@ -21,12 +22,18 @@ impl<'a, 'r> Context<'a, 'r> {
         self.get::<ServiceCache>()
     }
 
-    pub fn host(&self) -> Option<&str> {
-        self.request.headers().get_one("Host")
+    pub fn host(&self) -> Option<String> {
+        self.request.headers().get_one("Host").map(|host| {
+            format!("{}://{}", scheme(), host)
+        })
     }
 
-    pub fn path(&self) -> &str {
-        self.request.uri().path()
+    pub fn path(&self) -> String {
+        self.request.uri().path().to_string()
+    }
+
+    pub fn uri(&self) -> String {
```

Comment:
Had to add this for the cache key. Check in your redis-cli with `$ KEYS *` you will see that now keys containing path data with the correct cache. Right now with `http://localhost:8000/transactions/0x1230B3d59858296A31053C1b8562Ecf89A2f888b?next=limit%3D20%26offset%3D20` it should crash (I have implemented a fix for this in the local branch, that's related to #36 )